### ISSUE TITLE
Change IP address for NIC or add secondary - IPs not visible - 4.19

### ISSUE THREAD

**Issue by ccycv** [apache/cloudstack#8622]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.19
~~~

##### CONFIGURATION

~~~
Advanced Networking
~~~

##### SUMMARY

~~~
The free IPs are not visible in the UI when you attempt to change or assign secondary IP.
~~~
##### STEPS TO REPRODUCE

~~~

~~~
![image](https://github.com/apache/cloudstack/assets/23322009/d9167b0c-eb78-46df-ab39-9e9480f57dfe)

##### EXPECTED RESULTS

~~~
List the free IPs
~~~

##### ACTUAL RESULTS

~~~
Not visible 
~~~

**Comment by weizhouapache** [comment]
@ccycv 
looks like a ui bug
sometimes it works, sometimes not

**Comment by ccycv** [comment]
@weizhouapache 

Yes, I closed because after I did a check via API, the IPs where listed in UI also..

**Comment by weizhouapache** [comment]
@ccycv 
I have reopened this as I think this is a valid bug

**Comment by weizhouapache** [comment]
@ccycv 
can you check the management-server.log if there are similar erros like below ?
```
2024-02-07 10:59:11,110 ERROR [c.c.a.ApiServer] (qtp1146825051-20:ctx-74413901 ctx-b91ecc51) (logid:aa41ab63) unhandled exception executing api command: [Ljava.lang.String;@5a46b058
java.lang.NullPointerException
    at com.cloud.vm.NicVO.getInstanceId(NicVO.java:256)
    at com.cloud.api.ApiResponseHelper.addSystemVmInfoToIpResponse(ApiResponseHelper.java:995)
    at com.cloud.api.ApiResponseHelper.showVmInfoForSharedNetworks(ApiResponseHelper.java:1175)
    at com.cloud.api.ApiResponseHelper.createIPAddressResponse(ApiResponseHelper.java:1088)
    at org.apache.cloudstack.api.command.user.address.ListPublicIpAddressesCmd.execute(ListPublicIpAddressesCmd.java:196)
```

**Comment by ccycv** [comment]
@weizhouapache 

Looks like now I see the IPs when I list for change but for add secondary, same issue and i see this in logs :

`GET  nicId=41665ed7-bd8a-4e54-acc0-ac364e2f0d0e&keyword=&virtualmachineid=edf57c38-a206-4870-b67d-d32ab7680f81&command=listNics&response=json
2024-02-07 13:42:58,565 ERROR [c.c.a.ApiServer] (qtp249034932-20:ctx-4ac5f9b9 ctx-940c481b) (logid:e1869f4e) unhandled exception executing api command: [Ljava.lang.String;@14ae457a
java.lang.NullPointerException
        at com.cloud.vm.NicVO.getInstanceId(NicVO.java:256)
        at com.cloud.api.ApiResponseHelper.addSystemVmInfoToIpResponse(ApiResponseHelper.java:995)
        at com.cloud.api.ApiResponseHelper.addVmDetailsInIpResponse(ApiResponseHelper.java:982)
        at com.cloud.api.ApiResponseHelper.createIPAddressResponse(ApiResponseHelper.java:1042)
        at org.apache.cloudstack.api.command.user.address.ListPublicIpAddressesCmd.execute(ListPublicIpAddressesCmd.java:196)
        at com.cloud.api.ApiDispatcher.dispatch(ApiDispatcher.java:172)
        at com.cloud.api.ApiServer.queueCommand(ApiServer.java:782)
        at com.cloud.api.ApiServer.handleRequest(ApiServer.java:603)
        at com.cloud.api.ApiServlet.processRequestInContext(ApiServlet.java:347)
        at com.cloud.api.ApiServlet$1.run(ApiServlet.java:154)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:102)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52)
        at com.cloud.api.ApiServlet.processRequest(ApiServlet.java:151)
        at com.cloud.api.ApiServlet.doGet(ApiServlet.java:105)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:645)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
        at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1450)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:554)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:600)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1440)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:505)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:772)
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
        at org.eclipse.jetty.server.Server.handle(Server.java:516)
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487)
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
        at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:555)
        at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:410)
        at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:164)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
        at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
        at java.base/java.lang.Thread.run(Thread.java:829)`

**Comment by DaanHoogland** [comment]
fixed by #8624
